tests/integration/test_cluster_failover.py

```diff
@@ -21,44 +21,36 @@
 from time import sleep
 from unittest import TestCase
 
-import grakn_protocol.protobuf.cluster.database_pb2 as database_proto
-import grpc
-from grakn_protocol.protobuf.cluster.grakn_cluster_pb2_grpc import GraknClusterStub
-
-from grakn.client import GraknClient, SessionType, TransactionType
-from grakn.rpc.cluster.replica_info import ReplicaInfo
+from grakn.client import GraknClient, SessionType, TransactionType, DatabaseManagerCluster
 
 
 class TestClusterFailover(TestCase):
 
     def setUp(self):
         with GraknClient.cluster(["localhost:11729", "localhost:21729", "localhost:31729"]) as client:
             if "grakn" in client.databases().all():
-                client.databases().delete("grakn")
+                client.databases().get("grakn").delete()
             client.databases().create("grakn")
 
-    def get_primary_replica(self):
-        channel = grpc.insecure_channel("localhost:11729")
-        cluster_grpc_stub = GraknClusterStub(channel)
-        while True:
-            db_replicas_req = database_proto.Database.Replicas.Req()
-            db_replicas_req.database = "grakn"
+    def get_primary_replica(self, database_manager: DatabaseManagerCluster):
+        retry_num = 0
+        while retry_num < 10:
             print("Discovering replicas for database 'grakn'...")
-            res = cluster_grpc_stub.database_replicas(db_replicas_req)
-            dbs = ReplicaInfo.of_proto(res)
-            print("Discovered " + str([str(replica) for replica in dbs.replicas()]))
-            primary_replica = next(iter([replica for replica in dbs.replicas() if replica.is_primary()]), None)
-            if primary_replica:
-                return primary_replica
```

Comment:
As a consequence of the refactor, it's now much easier for users to get information like which Cluster node holds the current primary replica for a particular database.